### DISCUSSION TITLE
How to set blank prop for ProseA

### DISCUSSION THREAD

**Discussion by Benbinbin** [nuxt/content#1550] [Q&A] [answered]
I check out the [`ProseA.vue` Component](https://github.com/nuxt/content/blob/main/src/runtime/components/Prose/ProseA.vue) and find it can receive the `blank` prop

```vue
<script setup lang="ts">
defineProps({
  href: {
    type: String,
    default: ''
  },
  blank: {
    type: Boolean,
    default: false
  }
})
</script>

<template>
  <NuxtLink :href="href">
    <slot />
  </NuxtLink>
</template>
```

But how to set this prop in markdown, most of the time we create a link by this syntax `[link](www.example.com)` but there isn't other option to set the blank

**Comment by nobkd** [comment, marked as answer]
I think you have to use MDC syntax to be able define other props.
I'm relatively sure it's not possible to add other props when using the default Markdown syntax

**Reply by Benbinbin** [reply]
Refer to [`ProseCode` Component](https://github.com/nuxt/content/blob/main/src/runtime/components/Prose/ProseCode.vue) 

```vue
<script lang="ts">
import { defineComponent } from '#imports'

export default defineComponent({
  props: {
    code: {
      type: String,
      default: ''
    },
    language: {
      type: String,
      default: null
    },
    filename: {
      type: String,
      default: null
    },
    highlights: {
      type: Array as () => number[],
      default: () => []
    }
  }
})
</script>

<template>
  <slot />
</template>

<style>
pre code .line {
  display: block;
  min-height: 1rem;
}
</style>
```

It can accept the `code`, `language`, `filename`, `highlights` props

and we can set these props in markdown file like this

![block-code-with-more-info](https://user-images.githubusercontent.com/40909550/191894619-70a730a0-0ace-4ff2-91fa-36e47bf99c92.png)

* set the `language` prop value after three backquote
* the code block content is the `code` prop value
* set the `filename` prop value inside square brackets, like `[filename]` 
* set the `highlights` prop value inside the curly brackets, like `{2-4, 6}`

although the default `ProseCode` Component template doesn't utility the `language`, `filename`, `highlights` props, but I can [create a custom `ProseCode` Component](https://github.com/Benbinbin/BlogiNote/blob/main/components/content/ProseCode.vue)  to [use all these props](https://bloginote.benbinbin.com/article/tutorial/write-article#code).

So I think the Nuxt team should be set some ways to define the props at Markdown syntax with remark setting

**Reply by nobkd** [reply]
Soo. I looked into it again and found the code where the props are extracted.
I'll say it clear: I haven't thought of a way to implement the parsing for more props yourself.
I hope it helps, even if it's just a bit.

---

Here are the code references how it was done for the code block:

This code gives the first line (code block lang + everything else in the line) to a function:
https://github.com/nuxt/content/blob/main/src/runtime/markdown-parser/handler/code.ts#L8

The function parses the line with regex and returns the extracted values:
https://github.com/nuxt/content/blob/main/src/runtime/markdown-parser/handler/utils.ts#L8-L29

**Reply by Benbinbin** [reply]
thanks

**Reply by nobkd** [reply]
I thought of two ways. They're just thought experiments, but they might work out. 

It's probably possible using [transformers](https://content.nuxtjs.org/api/advanced#transformers) to convert your special links. I'm not exactly sure how to combine the default Markdown transformer with the new one (and if it's possible to set a parse order). 

Another way would be to use [before parse hook](https://content.nuxtjs.org/api/advanced#contentfilebeforeparse) to convert **your** link syntax to *MDC component syntax*, that then is parsed by the *default transformer*.

**Reply by Benbinbin** [reply]
Thanks I will think about adding a remark plugin or set the pre-process for markdown file in nuxt.

But now maybe the most easy way is to embed the `<a href="url" target="_blank">Link</a>` inside the markdown :smile: but I need to write the more  complex html syntax instead of the elegant markdown syntax :cry:

**Reply by nobkd** [reply]
Seems like this might interest you: https://github.com/nuxt/content/issues/1783#issuecomment-1367301722

**Reply by Benbinbin** [reply]
wow! That great! Thanks
